internal/mode/static/state/graph/backend_refs.go

```diff
@@ -116,33 +117,38 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
-	svc, port, err := getServiceAndPortFromRef(ref.BackendRef, sourceNamespace, services, refPath)
+	svcNsName, svcPort, err := getServiceAndPortFromRef(ref.BackendRef, sourceNamespace, services, refPath)
 	if err != nil {
 		backendRef = BackendRef{
-			Weight: weight,
-			Valid:  false,
+			SvcNsName:   svcNsName,
+			ServicePort: svcPort,
+			Weight:      weight,
+			Valid:       false,
 		}
 
 		cond := staticConds.NewRouteBackendRefRefBackendNotFound(err.Error())
 		return backendRef, &cond
 	}
 
 	backendRef = BackendRef{
-		Svc:    svc,
-		Port:   port,
-		Valid:  true,
-		Weight: weight,
+		SvcNsName:   svcNsName,
+		ServicePort: svcPort,
+		Valid:       true,
+		Weight:      weight,
 	}
 
 	return backendRef, nil
 }
 
+// getServiceAndPortFromRef can return an error and an empty v1.ServicePort in two cases:
```

Comment:
```suggestion
// getServiceAndPortFromRef extracts the NamespacedName of the Service and the port from a BackendRef.
It can return an error and an empty v1.ServicePort in two cases:
```

Reply:
I like this much better, the description on the method was missing, thanks